commons/zenoh-keyexpr/src/key_expr/intersect/mod.rs

```diff
@@ -38,22 +38,22 @@ pub trait Intersector<Left, Right> {
 }
 
 pub(crate) mod restriction {
-    use core::ops::Deref;
+    // use core::ops::Deref;
```

Comment:
Why not just remove them?

Reply:
There is an entire commented out module that relates to this code and I don't have context for why it's been commented out instead of left to git history.

Reply:
https://github.com/eclipse-zenoh/zenoh/blob/23c9b929e1101032a9d8745ab860b12e09211359/commons/zenoh-keyexpr/src/key_expr/intersect/mod.rs#L20-L28

It seems that these modules were deprecated before. I think you're right, I will delete it and add an issue so we can revisit it later.

Reply:
https://github.com/eclipse-zenoh/zenoh/issues/2083